mycli/main.py

```diff
@@ -656,21 +668,34 @@ def cli(database, user, host, port, socket, password, dbname,
 
     mycli.run_cli()
 
-def format_output(title, cur, headers, status, table_format):
+def format_output(title, cur, headers, status, table_format, expanded=False, max_width=None):
     output = []
     if title:  # Only print the title if it's not None.
         output.append(title)
     if cur:
         headers = [utf8tounicode(x) for x in headers]
-        if special.is_expanded_output():
+        if expanded:
             output.append(expanded_table(cur, headers))
         else:
-            output.append(tabulate(cur, headers, tablefmt=table_format,
-                missingval='<null>'))
+            tabulated, rows = tabulate(cur, headers, tablefmt=table_format,
+                missingval='<null>')
+            if (max_width and rows and
+                    content_exceeds_width(rows[0], max_width) and
+                    headers):
+                output.append(expanded_table(rows, headers))
```

Comment:
These `rows` are actually formatted with extra spacing. They potentially mess up the alignment. I'll push a commit that fixes this.